tunnels.go

```diff
@@ -0,0 +1,294 @@
+	if err != nil {
+		return Tunnel{}, err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return Tunnel{}, errors.Wrap(err, errUnmarshalError)
+	}
+
+	return argoDetailsResponse.Result, nil
+}
+
+// DeleteTunnel removes a single Argo tunnel.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-delete-cloudflare-tunnel
+func (api *API) DeleteTunnel(ctx context.Context, params TunnelDeleteParams) error {
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel/%s", params.AccountID, params.UUID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodDelete, uri, nil, tunnelV1Header())
+	if err != nil {
+		return err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return errors.Wrap(err, errUnmarshalError)
+	}
+
+	return nil
+}
+
+// CleanupTunnelConnections deletes any inactive connections on a tunnel.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-clean-up-cloudflare-tunnel-connections
+func (api *API) CleanupTunnelConnections(ctx context.Context, params TunnelCleanupParams) error {
+	if params.AccountID == "" {
+		return ErrMissingAccountID
+	}
+
+	if params.UUID == "" {
+		return errors.New("missing tunnel UUID")
+	}
+
+	uri := fmt.Sprintf("/accounts/%s/cfd_tunnel/%s/connections", params.AccountID, params.UUID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodDelete, uri, nil, tunnelV1Header())
+	if err != nil {
+		return err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return errors.Wrap(err, errUnmarshalError)
+	}
+
+	return nil
+}
+
+// TunnelToken gets the cloudflared token for a tunnel.
+//
+// API reference: https://api.cloudflare.com/#cloudflare-tunnel-get-cloudflare-tunnel-token
+func (api *API) TunnelToken(ctx context.Context, params TunnelTokenParams) error {
+	if params.AccountID == "" {
+		return ErrMissingAccountID
+	}
+
+	if params.UUID == "" {
+		return errors.New("missing tunnel UUID")
+	}
+
+	uri := fmt.Sprintf("accounts/%s/cfd_tunnel/%s/token", params.AccountID, params.UUID)
+
+	res, err := api.makeRequestContextWithHeaders(ctx, http.MethodDelete, uri, nil, tunnelV1Header())
+	if err != nil {
+		return err
+	}
+
+	var argoDetailsResponse TunnelDetailResponse
+	err = json.Unmarshal(res, &argoDetailsResponse)
+	if err != nil {
+		return errors.Wrap(err, errUnmarshalError)
+	}
+
+	return nil
+}
+
+// The early implementation of Tunnel endpoints didn't conform to the V4
+// API standard response structure. This has been remedied going forward however
+// to support older clients this isn't yet the default. An explicit `Accept`
+// header is used to get the V4 compatible version.
+func tunnelV1Header() http.Header {
+	header := make(http.Header)
+	header.Set("Accept", "application/json;version=1")
```

Comment:
The `version=1` is no longer needed in the new APIs used in this file

Reply:
🎉 nice!